### PR TITLE
Update zstd to 1.4.4

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **zstd/1.4.4**

- This recipe had many "mistakes" according Conan Center hook.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zstd/all/conanfile.py [hunk 1]
```diff
@@ -35,7 +30,7 @@ def config_options(self):
 
     def configure(self):
         del self.settings.compiler.libcxx
-        del self.settings.compiler.libstd
```
Comment (SSE4) [inline recipes/zstd/all/conanfile.py]: interesting, we might need to improve hook to detect such misspellings

Comment (uilianries) [reply recipes/zstd/all/conanfile.py]: not bad idea!

Comment (jgsogo) [reply recipes/zstd/all/conanfile.py]: Maybe it should be Conan itself?

Comment (uilianries) [reply recipes/zstd/all/conanfile.py]: @jgsogo Yeah, why not. If a setting is being delete and is not listed in settings.yml, then we can raise an exception.

**Pull request merged**